tests/osaka/eip7212_secp256r1_precompiles/conftest.py

```diff
@@ -0,0 +1,161 @@
+    if vector_gas_value is not None:
+        assert vector_gas_value == Spec.P256VERIFY_GAS, (
+            f"Calculated gas {vector_gas_value} != Vector gas {Spec.P256VERIFY_GAS}"
+        )
+    return Spec.P256VERIFY_GAS
+
+
+@pytest.fixture
+def precompile_gas_modifier() -> int:
+    """
+    Modify the gas passed to the precompile, for testing purposes.
+
+    By default the call is made with the exact gas amount required for the given opcode,
+    but when this fixture is overridden, the gas amount can be modified to, e.g., test
+    a lower amount and test if the precompile call fails.
+    """
+    return 0
+
+
+@pytest.fixture
+def call_opcode() -> Op:
+    """
+    Type of call used to call the precompile.
+
+    By default it is Op.CALL, but it can be overridden in the test.
+    """
+    return Op.CALL
+
+
+@pytest.fixture
+def call_contract_post_storage() -> Storage:
+    """
+    Storage of the test contract after the transaction is executed.
+    Note: Fixture `call_contract_code` fills the actual expected storage values.
+    """
+    return Storage()
+
+
+@pytest.fixture
+def call_succeeds() -> bool:
+    """
+    By default, depending on the expected output, we can deduce if the call is expected to succeed
+    or fail.
+    """
+    return True
+
+
+@pytest.fixture
+def call_contract_code(
+    precompile_address: int,
+    precompile_gas: int,
+    precompile_gas_modifier: int,
+    expected_output: bytes | SupportsBytes,
+    call_succeeds: bool,
+    call_opcode: Op,
+    call_contract_post_storage: Storage,
+) -> Bytecode:
+    """Code of the test contract."""
+    expected_output = bytes(expected_output)
+    assert call_opcode in [Op.CALL, Op.CALLCODE, Op.DELEGATECALL, Op.STATICCALL]
+    value = [0] if call_opcode in [Op.CALL, Op.CALLCODE] else []
+
+    code = Op.CALLDATACOPY(0, 0, Op.CALLDATASIZE()) + Op.SSTORE(
+        call_contract_post_storage.store_next(call_succeeds),
+        call_opcode(
+            precompile_gas + precompile_gas_modifier,
+            precompile_address,
+            *value,
+            0,
+            Op.CALLDATASIZE(),
+            0,
+            0,
+        )
+        + Op.SSTORE(
+            call_contract_post_storage.store_next(len(expected_output)), Op.RETURNDATASIZE()
+        ),
+    )
+    if call_succeeds:
+        # Add integrity check only if the call is expected to succeed.
+        code += Op.RETURNDATACOPY(0, 0, Op.RETURNDATASIZE()) + Op.SSTORE(
+            call_contract_post_storage.store_next(keccak256(expected_output)),
+            Op.SHA3(0, Op.RETURNDATASIZE()),
+        )
+    return code
+
+
+@pytest.fixture
+def call_contract_address(pre: Alloc, call_contract_code: Bytecode) -> Address:
+    """Address where the test contract will be deployed."""
+    return pre.deploy_contract(call_contract_code)
+
+
+@pytest.fixture
+def sender(pre: Alloc) -> EOA:
+    """Sender of the transaction."""
+    return pre.fund_eoa(1_000_000_000_000_000_000)
```

Comment:
```suggestion
    return pre.fund_eoa()
```
The default amount should be enough, otherwise we can take a look if any of the tests is using too much gas.